scripts/builtin/csplineDS.dml

```diff
@@ -0,0 +1,183 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+#
+# THIS SCRIPT SOLVES CUBIC SPLINE INTERPOLATION USING THE DIRECT SOLVER
+#
+# INPUT PARAMETERS:
+# --------------------------------------------------------------------------------------------
+# NAME   TYPE            DEFAULT  MEANING
+# --------------------------------------------------------------------------------------------
+# X      Matrix[Double]    ---     1-column matrix of x values knots
+# Y      Matrix[Double]    ---     1-column matrix of corresponding y values knots
+# inp_x  Double            ---     the given input x, for which the cspline will find predicted y.
+#
+# --------------------------------------------------------------------------------------------
+# OUTPUT: 
+# pred_y Matrix[Double]    ---     Predicted value
+# K      Matrix[Double]    ---     Matrix of k parameters 
+#
+
+#Assumptions:
+# - The inputs xs are monotonically increasing,
+# - there is no duplicates points in x
```

Comment:
again assumptions in docs above, not here. We use the above table to generate code and comments elsewhere, therefore this would be ignored.